trade-finance-manager-api/src/v1/controllers/user/user.controller.upsert-tfm-user-from-entra-id-user.test.ts

```diff
@@ -0,0 +1,54 @@
+import { when, resetAllWhenMocks } from 'jest-when';
+import { anEntraIdUser, AuditDetails, EntraIdUser } from '@ukef/dtfs2-common';
+import { generateSystemAuditDetails } from '@ukef/dtfs2-common/change-stream';
+import { mapUserData } from './helpers/mapUserData.helper';
+import { UpsertTfmUserFromEntraIdUserResponse, UserService } from '../../services/user.service';
+import { upsertTfmUserFromEntraIdUser } from './user.controller';
+import { userServiceMockResponses } from '../../../../test-helpers';
+import { TfmSessionUser } from '../../../types/tfm-session-user';
+
+jest.mock('../../services/user.service', () => ({
+  __esModule: true,
+  UserService: {
+    upsertTfmUserFromEntraIdUser: jest.fn(),
+  },
+}));
+
+describe('user controller', () => {
+  describe('upsertTfmUserFromEntraIdUser', () => {
+    let entraIdUser: EntraIdUser;
+    let auditDetails: AuditDetails;
+
+    let upsertTfmUserFromEntraIdUserResponse: UpsertTfmUserFromEntraIdUserResponse;
+    let mappedUserDetails: TfmSessionUser;
+
+    beforeEach(() => {
+      jest.resetAllMocks();
+      resetAllWhenMocks();
+      entraIdUser = anEntraIdUser();
+      auditDetails = generateSystemAuditDetails();
+
+      upsertTfmUserFromEntraIdUserResponse = userServiceMockResponses.anUpsertTfmUserFromEntraIdUserResponse();
+      mappedUserDetails = mapUserData(upsertTfmUserFromEntraIdUserResponse);
+      // eslint-disable-next-line @typescript-eslint/unbound-method
```

Comment:
can this be removed

Reply:
A very interesting question -- I've got a little time to have a look at this. The reason why this doesn't matter is because it is a static class (there is a ESLint rule we could change to allow static classes unbound methods).

It seems an odd interaction between jest and classes -- We cannot move to DI in the constructor, which would avoid this issue all together, due to the heavy use of this controller service

Reply:
That was good fun -- Updated